replication-api-impl/src/main/java/com/connexta/ion/replication/api/impl/data/SiteIndexImpl.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.ion.replication.api.impl.data;
+
+import com.connexta.ion.replication.api.data.SiteIndex;
+import java.util.Date;
+import java.util.UUID;
+import org.apache.commons.lang3.Validate;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.solr.core.mapping.Indexed;
+import org.springframework.data.solr.core.mapping.SolrDocument;
+
+/** Simple implementation of SiteIndex. */
+@SolrDocument(collection = "site_index")
+public class SiteIndexImpl implements SiteIndex {
+
+  private static final int CURRENT_VERSION = 1;
+
+  @Id private final String id;
+
+  @Indexed(name = "version")
+  private final int version;
+
+  @Indexed(name = "modified_since")
+  private final Date modifiedSince;
+
+  @Indexed(name = "site_id")
+  private final String siteId;
+
+  /**
+   * Creates a new SiteIndex with the current version and a random UUID.
+   *
+   * @param modifiedSince the date to query resources that have been modified since
+   * @param siteId the site id this index applies to
+   */
+  public SiteIndexImpl(Date modifiedSince, String siteId) {
+    this(UUID.randomUUID().toString(), CURRENT_VERSION, modifiedSince, siteId);
+  }
+
+  /**
+   * Creates a new SiteIndex.
+   *
+   * @param id the unique id of this index
+   * @param version the version of this index
+   * @param modifiedSince the date to query resources that have been modified since
+   * @param siteId the site id this index applies to
+   */
+  public SiteIndexImpl(String id, int version, Date modifiedSince, String siteId) {
```

Comment:
Yeah it will use the all args ctor, but I cannot have the other ctor actually for solr to be able to recognize the all args without telling solr which ctor to use. So I may change the first ctor to a factory method. 
Solr will match the property name to the ctor name.
This implementation should be enforcing the interface contract right? And then the pojo + manager will take care of making sure we convert and detect objects correctly.